### DISCUSSION TITLE
API Key revoked at mailchimp.

### DISCUSSION THREAD

**Discussion by AndreTooGood** [community/community#51672] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hi there. I recently hosted my web app on cyclic.sh but there is an issue towards the end, for the "re-enabling" of my API key which has of course been revoked as the files were posted on a public repository here on GitHub. How do I exactly re-enable it, as there doesn't seem to be any option to do so. I feel I need to generate another API key, but then everything will start breaking from there on. Any help is highly appreciated.

**Comment by ZaPPEr001K** [comment, marked as answer]
same problem. Was doing a course, which suggested to upload on Heroku, which since is no longer free, i tried uploading at Cyclic, and now i saw the API key being revoked.

**Reply by AndreTooGood** [reply]
Angela's course on Udemy if I'm correct. I figured out how to get it done. Just generate a new API key, and edit the app.js file on the repository, while it is private of course (if it's public it will get revoked again so make sure), and replace the revoked key with the brand new one. Cyclic will re-deploy the site and you are good to go! It was quite simple and straightforward haha. I was very skeptical to do, but, it's always a good idea to try things out, never hurts!

**Reply by ajaykumar-847** [reply]
Thank you so much Andre. I was searching for a solution and could not find it anywhere. Your answer helped me a lot.

**Reply by ALPHA-369** [reply]
> Angela's course on Udemy if I'm correct. I figured out how to get it done. Just generate a new API key, and edit the app.js file on the repository, while it is private of course (if it's public it will get revoked again so make sure), and replace the revoked key with the brand new one. Cyclic will re-deploy the site and you are good to go! It was quite simple and straightforward haha. I was very skeptical to do, but, it's always a good idea to try things out, never hurts!

Thanks a lot Andre, was just facing same issue now

**Reply by ShivamDeulkar** [reply]
> Angela's course on Udemy if I'm correct. I figured out how to get it done. Just generate a new API key, and edit the app.js file on the repository, while it is private of course (if it's public it will get revoked again so make sure), and replace the revoked key with the brand new one. Cyclic will re-deploy the site and you are good to go! It was quite simple and straightforward haha. I was very skeptical to do, but, it's always a good idea to try things out, never hurts!

Thanks Andre, it helped me a lot

**Reply by pattysue1** [reply]
~~~~

**Reply by ALPHA-369** [reply]
> Angela's course on Udemy if I'm correct. I figured out how to get it done. Just generate a new API key, and edit the app.js file on the repository, while it is private of course (if it's public it will get revoked again so make sure), and replace the revoked key with the brand new one. Cyclic will re-deploy the site and you are good to go! It was quite simple and straightforward haha. I was very skeptical to do, but, it's always a good idea to try things out, never hurts!

Genius idea but still not best practice

**Comment by ALPHA-369** [comment]
I was facing same issue yesterday, used a new API key in my  repo and it worked but got revoked sometime later. Why this is happening is because your API key is meant to be private as anyone who access to API key can actually do some damage. So the best practice is to create a .env file where you store your API key and other sensitive data. You can then request for this data using process.env.(whatever name you used to store your API key) Note that you have to install the dotenv package and require in your code. Create a .gitignore file and add the .env file to it. What this does is that when you push your files to your repo, it doesn't include the .env file therefore no one has access to it.
The downside to this is that when you use domain service like render which uses the data in your github repo, your API_key name becomes undefined. I use render and when creating a web service, there's a provision to create an environment variable or a secret file such as .env . So you can go ahead and declare your API_key as an enviroment variable or simply create a .env file and paste your API_key in there. If my answer is not concise enough, here's a link to video that helped me https://youtu.be/17UVejOw3zA

**Reply by AndreTooGood** [reply]
> I was facing same issue yesterday, used a new API key in my  repo and it worked but got revoked sometime later. Why this is happening is because your API key is meant to be private as anyone who access to API key can actually do some damage. So the best practice is to create a .env file where you store your API key and other sensitive data. You can then request for this data using process.env.(whatever name you used to store your API key) Note that you have to install the dotenv package and require in your code. Create a .gitignore file and add the .env file to it. What this does is that when you push your files to your repo, it doesn't include the .env file therefore no one has access to it.
> The downside to this is that when you use domain service like render which uses the data in your github repo, your API_key name becomes undefined. I use render and when creating a web service, there's a provision to create an environment variable or a secret file such as .env . So you can go ahead and declare your API_key as an enviroment variable or simply create a .env file and paste your API_key in there. If my answer is not concise enough, here's a link to video that helped me https://youtu.be/17UVejOw3zA

Will consider this. Thanks!

**Reply by ummulzehra** [reply]
can you share the piece cod code where you changed it because i am changing it and it is giving error

**Reply by Aneeshaik** [reply]
> I was facing same issue yesterday, used a new API key in my repo and it worked but got revoked sometime later. Why this is happening is because your API key is meant to be private as anyone who access to API key can actually do some damage. So the best practice is to create a .env file where you store your API key and other sensitive data. You can then request for this data using process.env.(whatever name you used to store your API key) Note that you have to install the dotenv package and require in your code. Create a .gitignore file and add the .env file to it. What this does is that when you push your files to your repo, it doesn't include the .env file therefore no one has access to it. The downside to this is that when you use domain service like render which uses the data in your github repo, your API_key name becomes undefined. I use render and when creating a web service, there's a provision to create an environment variable or a secret file such as .env . So you can go ahead and declare your API_key as an enviroment variable or simply create a .env file and paste your API_key in there. If my answer is not concise enough, here's a link to video that helped me https://youtu.be/17UVejOw3zA

Thank you so muxh
